From 7bf494eb0838f328c1247f60deb637552c1fb170 Mon Sep 17 00:00:00 2001
From: Colin Walters <walters@verbum.org>
Date: Tue, 11 Oct 2016 11:28:39 -0400
Subject: [PATCH] priorities: Do read crypto policy files with mtime of zero

In a default Fedora Atomic Host installation,
`/etc/crypto-policies/backends/gnutls.config` is a symlink to the
default in `/usr/share/`.  On an OSTree-managed system, files in
`/usr` have an mtime of zero (to help deduplication).

The simple fix here is to still try to read the first time, even if
the file has an mtime of zero.
---
 lib/priority.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/priority.c b/lib/priority.c
index 0fa38f9..880fa58 100644
--- a/lib/priority.c
+++ b/lib/priority.c
@@ -945,7 +945,8 @@ static void _gnutls_update_system_priorities(void)
 		return;
 	}
 
-	if (sb.st_mtime == system_priority_last_mod) {
+	if (system_priority_buf != NULL &&
+	    sb.st_mtime == system_priority_last_mod) {
 		_gnutls_debug_log("system priority %s has not changed\n",
 				  system_priority_file);
 		return;
-- 
2.7.4

